services/github/github-total-star.service.js

```diff
@@ -0,0 +1,259 @@
+</p>
+`
+const orgDocumentation = `${commonDocumentation}
+<p>
+  <b>Note:</b> ${customDocumentation}
+</p>`
+
+const pageInfoSchema = Joi.object({
+  hasNextPage: Joi.boolean().required(),
+  endCursor: Joi.string().allow(null).required(),
+}).required()
+
+const nodesSchema = Joi.array()
+  .items(
+    Joi.object({
+      stargazers: Joi.object({
+        totalCount: nonNegativeInteger.required(),
+      }).required(),
+    })
+  )
+  .default([])
+
+const repositoriesSchema = Joi.object({
+  pageInfo: pageInfoSchema,
+  nodes: nodesSchema,
+}).required()
+
+const schema = Joi.object({
+  data: Joi.alternatives(
+    Joi.object({
+      user: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required(),
+    Joi.object({
+      organization: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required()
+  ).required(),
+}).required()
+
+const userQuery = gql`
+  query fetchStars(
+    $user: String!
+    $nextCursor: String
+    $affiliations: [RepositoryAffiliation]!
+  ) {
+    user(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        affiliations: $affiliations
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const orgQuery = gql`
+  query fetchStars($user: String!, $nextCursor: String) {
+    organization(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+const affiliationsAllowedValues = [
+  'OWNER',
+  `COLLABORATOR`,
+  'ORGANIZATION_MEMBER',
+]
+/**
+ * Validates affiliations should contain combination of allowed values in any order.
+ *
+ * @param {string} value affiliation current value
+ * @param {*} helpers object to construct custom error
+ *
+ * @returns {string} valiadtion error or value unchanged
+ */
+const validateAffiliations = (value, helpers) => {
+  const values = value.split(',')
+  if (values.some(e => !affiliationsAllowedValues.includes(e))) {
+    return helpers.error('any.invalid')
+  }
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  org: Joi.valid('').optional(),
+  // TODO add regex
+  affiliations: Joi.string().default('OWNER').custom(validateAffiliations),
+}).required()
+
+module.exports = class TotalStarService extends GithubAuthV4Service {
+  static get defaultLabel() {
+    return 'Total Stars'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'github/users',
```

Comment:
I don't think `users` is the right noun to have here (see our [badge url conventions](https://github.com/badges/shields/blob/master/doc/badge-urls.md#badge-url-conventions) for more info). Something like `stars` or `total-stars` would be better.

Personally I think `stars` would suffice and be a better route match with the existing repo stars, and there's no concerns about route conflicts (the other requires an owner and repo), and there's query params used to toggle between an org and user to differentiate between those two

Reply:
Done